### PR TITLE
dev: chg: solve vulns

### PR DESCRIPTION
**Pull request by marcello33**
This PR bumps some dependencies to solve reported vulnerabilities, and change code in some places to make it compatible with new libraries.

### CODE REVIEW

File: crypto/secp256k1/secp256k1_nocgo.go [hunk 1]
```diff
@@ -52,24 +52,3 @@ func (pubKey PubKeySecp256k1) VerifyBytes(msg []byte, sigStr []byte) bool {
 	return ethCrypto.VerifySignature(pubKey[:], hash, sigStr[:64])
 	// return signature.Verify(crypto.Sha256(msg), pub)
 }
-
-// Read Signature struct from R || S. Caller needs to ensure
-// that len(sigStr) == 64.
-func signatureFromBytes(sigStr []byte) *secp256k1.Signature {
```
Comment (Raneet10) [inline crypto/secp256k1/secp256k1_nocgo.go:58]: Are we sure about removing this ?

File: crypto/secp256k1/secp256k1_nocgo.go [hunk 2]
```diff
@@ -52,24 +52,3 @@ func (pubKey PubKeySecp256k1) VerifyBytes(msg []byte, sigStr []byte) bool {
...
-	return &secp256k1.Signature{
-		R: new(big.Int).SetBytes(sigStr[:32]),
-		S: new(big.Int).SetBytes(sigStr[32:64]),
-	}
-}
-
-// Serialize signature to R || S.
-// R, S are padded to 32 bytes respectively.
-func serializeSig(sig *secp256k1.Signature) []byte {
```
Comment (Raneet10) [inline crypto/secp256k1/secp256k1_nocgo.go:67]: Same here.

File: crypto/secp256k1/secp256k1_nocgo_test.go [hunk 3]
```diff
@@ -6,40 +7,8 @@ import (
 	"testing"
 
 	"github.com/stretchr/testify/require"
-	"github.com/tendermint/tendermint/crypto/secp256k1"
 )
 
-// Ensure that signature verification works, and that
-// non-canonical signatures fail.
-// // Note: run with CGO_ENABLED=0 or go test -tags !cgo.
-func TestSignatureVerificationAndRejectUpperS(t *testing.T) {
```
Comment (Raneet10) [inline crypto/secp256k1/secp256k1_nocgo_test.go:15]: Seems innocuous but curious why we remove this test. Is it only concerning `nocgo` implementations ?

**Review by Raneet10 — APPROVED** [review]
LGTM mostly, just some minor questions.

**Comment by marcello33** [comment]
Thanks for the comments @Raneet10 

I deleted those portions of code because the implementation with the new package requires some effort and we need to release the patch asap. It is safe as both these functions (`serializeSig` and `signatureFromBytes`) are only used in `TestSignatureVerificationAndRejectUpperS`. The complete test suite runs successfully.

**Review by ssandeep — APPROVED** [review]

**Pull request merged**